osi_vehicle.proto

```diff
@@ -0,0 +1,692 @@
+        //
+        optional double stw_friction = 4;
+
+        // Hands Off Detection.
+        //
+        optional bool are_hands_off = 5;
+    }
+
+    //
+    // \brief Interface to the vehicle-model.
+    // The focus here is on the wheels.
+    // It is made usage of the wheel-message to shorten the code.
+    //
+    message VehicleWheels
+    {
+        // Contains the physical description of the front-left wheel.
+        //
+        optional Wheel wheel_front_left = 1;
+
+        // Contains the physical description of the front-right wheel.
+        //
+        optional Wheel wheel_front_right = 2;
+
+        // Contains the physical description of the rear-left wheel.
+        //
+        optional Wheel wheel_rear_left = 3;
+
+        // Contains the physical description of the rear-right wheel.
+        //
+        optional Wheel wheel_rear_right = 4;
+    }
+
+    //
+    // \brief Interface to the vehicle-model.
+    // The focus here is on the physical description of a wheel.
+    //
+    message Wheel
+    {
+        // Contains the friction-coefficient of each wheel.
+        // Dimensionless.
+        //
+        // Unit: []
+        //
+        optional double friction_coefficient = 1;
+
+        // Contains the x, y and z-coordinate of the contact point of 
+        // the wheel, so that walking, torsion and deflation can be visualized.
+        // relative to the center of the wheel.
+        //
+        // Unit: [m]
+        //
+        optional Vector3d contact_point = 2;
+
+        // Contains the rotational speed of each wheel per second.
+        //
+        // Unit: [rad/s]
+        //
+        optional double rotational_speed = 3;
+
+        // Contains the steering angle of each wheel.
+        //
+        // Unit: [rad]
+        //
+        optional double steeringangle = 4;
+
+        // Contains the camber of each wheel.
+        // 
+        // Negative camber if the bottom of the wheel is farther out than the
+        // top.
+        // For more information: https://en.wikipedia.org/wiki/Camber_angle.
+        //
+        // Unit: [rad]
+        //
+        optional double camber = 5;
+
+        // Contains the tirepressure of each tire.
+        //
+        // Unit: [Pa]
+        // 
+        optional double tirepressure = 6;
+
+        // Contains the springdeflection in z-direction for each wheel.
+        //
+        // Unit: [mm]
+        //
+        optional double springdeflection = 7;
+
+        //Contains the relativ position of the wheel to the center of the car
+        //
+        //Unit: [m]
+        //
+        optional Vector3d position = 8;
+    
+        //Contains the relativ orientation of the wheel to the center of the car
+        //As the rotation of the wheel is also controlled by this value, 
+        //
+        //Unit: [rad]
+        //
+        optional Orientation3d orientation = 9;
+
+        //Contains the absolute (longitunal) slip of the tire
+        //0-100 percent
```

Comment:
Look at the comment rules here: https://opensimulationinterface.github.io/osi-documentation/osi/commenting.html